python/ray/monitor.py

```diff
@@ -534,6 +548,37 @@ def update_local_scheduler_map(self):
                 or local_scheduler_info["NodeManagerAddress"]).split(":")[0]
             self.local_scheduler_id_to_ip_map[client_id] = ip_address
 
+    def _maybe_flush_gcs(self):
+        """Experimental: issue a flush request to the GCS.
+
+        The purpose of this feature is to control GCS memory usage.
+
+        To activate this feature, Ray must be compiled with the flag
+        RAY_USE_NEW_GCS set, and Ray must be started at run time with the flag
+        as well.
+        """
+        if not self.issue_gcs_flushes:
+            return
+        if self.gcs_flush_policy is None:
+            serialized = self.redis.get("gcs_flushing_policy")
+            if serialized is None:
+                # Client has not set any policy; by default flushing is off.
+                return
+            self.gcs_flush_policy = pickle.loads(serialized)
+
+        if not self.gcs_flush_policy.should_flush(self.redis):
+            return
+
+        max_entries_to_flush = self.gcs_flush_policy.num_entries_to_flush()
+        num_flushed = self.redis.execute_command(
+            "HEAD.FLUSH {}".format(max_entries_to_flush))
```

Comment:
Why not expose a thin wrapper around this command? That would give users the maximum flexibility.

Reply:
I think it's important/beneficial to emphasize the automatic aspect of this feature.  This will encourage the users to play with the simple policy first.  We want to develop that to a mature enough degree.

Reply:
Sure, but then for it to be useful it needs to work perfectly. If we expose this command then the bar is lower and the user will usually be able to work around problems.